### DISCUSSION TITLE
Foundations for Storyboard Feature Development

### DISCUSSION THREAD

**Discussion by llloveletter** [tahoma2d/tahoma2d#1585] [Ideas]
This discussion is mainly to reiterate the goals proposed in the [initial PR](https://github.com/opentoonz/opentoonz/pull/3665) for the Preproduction Board, and to further solidify what foundations to lay for storyboarding feature development. The original goals are as follows:

- [ ] 1. Allow to change order of scenes
- [ ] 2. New button "Add scene named..." which asks for scene name
- [ ] 3. Allow to playback scenes together, in sequence
- [ ] 4. Option to show scenes horizontally, like in [WonderUnit's Storyboarder](https://wonderunit.com/storyboarder/)
- [ ] 5. Option to show scene length (when horizontal mode enabled)
- [ ] 6. Allow to change scene length (it will change playback range of scene)
- [ ] 7. Allow to duplicate scenes
- [ ] 8. Add background soundtrack when scenes show in horizontal mode
- [ ] 9. Export storyboard as PDF

Tahoma is amazing for anything beyond layout, but starkly lacking when it comes to storyboard functionality; it would be a shame not to develop this aspect of the software. Below I'd like to explore what other crucial features I'd like to see in Tahoma, and encourage you to give your own input on the matter.

---

### What I'd Like to See

The initial PR notes it's goal to provide functionality similar to [Storyboarder](https://wonderunit.com/storyboarder/), or [Story Planner](https://youtu.be/hZcYTu8PfRM), but I would be remiss not to mention [Griffith](https://www.youtube.com/watch?v=RWdw9rDRf7A) for consideration in feature development. I believe Griffith is leading the way in storyboarding software, and that's largely to do with it's compliance to the well established ekonte format of storyboards. Seeing as Griffith and OpenToonz both fundamentally cater to professional Japanese animation, I feel the need to better voice my favor for the ekonte format's role in future design.

There are two features from Griffith which could prove useful short-term additions to the previously listed goals:

- [In-canvas Frames](https://youtu.be/RWdw9rDRf7A?t=459): Viewer option to arrange neighboring frames vertically above and below the camera, just as in the ekonte format. For use in maintaining continuity, just as the Onion Skin feature. (Note that OT/T2D camera isn't static like Griffith's). 
- [Margin Notes](https://youtu.be/RWdw9rDRf7A?t=501): A virtual space on the camera or canvas margins reserved for the sole use of handwritten notes. This mainly serves as a quality of life feature, but can also briefly act as a 'captioning' system, where the most common notes associated with ekonte sheets might find themself (action, dialogue, sound, scene/cut).

With some modification, the Preproduction Panel can also serve as an [Overview Panel](https://youtu.be/RWdw9rDRf7A?t=520), but the relation between scenes and frames gets a bit mixed in translation between software.

Most discussions of OT/T2D are in the Discords of each respectively, but I want some concrete discussion on storyboard development that I won't have to go fishing for a couple months later. Also, just to cap things off, I encourage you to watch the video presentation on [Griffith](https://www.youtube.com/watch?v=RWdw9rDRf7A) just to understand why I'm so adamant of it's design.

**Comment by artisteacher** [comment]
I haven't watched the video, but I do think the preproduction board would benefit from some sort of note/caption feature.

**Comment by andeon** [comment]
I really liked the [Storypencil tool](https://www.youtube.com/watch?v=b25kfE6qd_c). I was wondering if it would be possible to do something like that by working with the timeline and levels, but when editing the level, all instances of the viewer and combo viewer are synchronized. I tried using the flipbook together with scene cast, but I always have to use the view to reopen the level in order to update it. If it synchronized the level, it would be interesting for certain situations, and it would be even better if it were possible to add all the levels. This way, we would have a way to view the raw form while editing. I've been using the new scene depth feature to separate things at different times in the animation. I would really like how this could be dynamic, as it is in Blender. The flexibility is so great that the storyboard could even quickly turn into an animatic, if desired."

**Comment by matitanimata** [comment]
Since I've long dreamed of a version of Tahoma — or at least a tool within it — that could make it suitable for professional-level storyboard production, I’d like to offer some feedback on the Preproduction Board feature currently being developed. 
While I understand its purpose — helping manage multiple shots in a simplified way — I feel that the current approach of generating separate ⁠ .tnz ⁠ files per shot may actually introduce more fragmentation than flexibility.

This method lacks a global timeline, makes animatic editing difficult, and complicates both creative and technical workflows, especially when trying to make overall timing and audio adjustments or preview the full sequence.

▶️ My Alternative Workflow (based on nested sub-scenes):

I’ve been developing a different workflow that keeps everything structured within a single main scene using sub-scenes, which offers a much more integrated and flexible experience — closer to how tools like Toon Boom Storyboard Pro (and storyplanner, which many years ago I helped develop as a consultant) operate. Here's how it works:

1.⁠ ⁠Top Level (Layer 1) – The Storyboard Layout:
 This layer contains the storyboard pages. Each panel on the page references a sub-scene that contains the actual animatic. I export this level as a sequence of images (PNG/JPG) to create a storyboard PDF.

2.⁠ ⁠Middle Level (Layer 2) – The Full Animatic Scene:  
   This is a sub-scene containing the full timeline of the animatic — with audio and all shots inserted as sub-scenes. I edit overall timing here, sync audio, and manage the flow of the full piece. This is where I export the final animatic as a video (MOV).

3.⁠ ⁠Bottom Level (Layer 3) – The Individual Shot Sub-scenes:  
   Each of these sub-scenes contains one shot: poses, camera movements, and trimmed audio specific to that shot.  I export each as a standalone ⁠ .tnz ⁠ using “Save Sub-scene As” — to be used later as layout.

🎯 Why this is useful:
•⁠  ⁠Everything stays in one file structure: easy to manage, back up, and edit.
•⁠  ⁠You can preview the full animatic without jumping between files.
•⁠  ⁠You can generate all three key storyboard deliverables:
  - Storyboard PDF from Layer 1
  - Animatic video from Layer 2
  - Individual shot ⁠ .tnz ⁠ layouts from Layer 3
  
This nested approach allows for a non-destructive, scalable workflow that supports both storytelling and production.

It’s a bit of a layered method, but it works really well in production — no need to jump between dozens of separate files.

I hope this workflow can inspire future development toward a storyboard-oriented version of Tahoma2D, where this kind of structure could be integrated more natively and with a simpler UI.

Happy to share a sample project if useful.
